sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -1249,6 +1250,150 @@ private void SetNameFieldsIfNull()
         }
         #endregion Parallel Download
 
+        #region OpenRead
+        /// <summary>
+        /// Opens a stream for reading from the blob.  The stream will only download
+        /// the blob as the stream is read from.
+        /// </summary>
+        /// <param name="position">
+        /// The position within the blob to begin the stream.
+        /// Defaults to the beginning of the blob.
+        /// </param>
+        /// <param name="bufferSize">
+        /// The buffer size to use when the stream downloads parts
+        /// of the blob.  Defaults to 1 MB.
+        /// </param>
+        /// <param name="conditions">
+        /// Optional <see cref="BlobRequestConditions"/> to add conditions on
+        /// the download of the blob.
+        /// </param>
+        /// <returns>
+        /// Returns a stream that will download the blob as the stream
+        /// is read from.
+        /// </returns>
+#pragma warning disable AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+#pragma warning disable AZC0015 // Unexpected client method return type.
+        public virtual Stream OpenRead(
+#pragma warning restore AZC0015 // Unexpected client method return type.
+#pragma warning restore AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+            long position = 0,
+            int bufferSize = Constants.DefaultStreamingDownloadSize,
+            BlobRequestConditions conditions = default)
```

Comment:
I'm not sure how I would call Download() with the ETag specified.  In the current form, here is my download lamda: `private readonly Func<HttpRange, TRequestConditions, bool, bool, CancellationToken, Task<Response<TDownloadInfo>>> _downloadInternalFunc;`.  How specify `IfMatch` on  `TRequestConditions`?

Reply:
Since `IfMatch` comes from `RequestConditions` and all your different types probably derive from it, you could add a `where TRequestConditions : RequestConditions` type constraint.

Reply:
`ShareFileRequestConditions` does not extend `RequestConditions`.  How do we move forward?

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareFileRequestConditions.cs

Reply:
Could we change it to something more like this then?

```C#
if (response.Headers.ETag != null &&
    conditions is RequestConditions c)
{
    c.IfMatch = response.Headers.ETag;
}
```

Reply:
I'm a bit confused, what would happen in the case where conditions is ShareFileRequestConditions which doesn't extend RequestConditions?

Reply:
You'd do nothing for `ShareFileRequestConditions`, but I don't think you have a choice for Files [because it doesn't support `If-Match`.](https://docs.microsoft.com/en-us/rest/api/storageservices/get-file#request-headers)

Reply:
Added in next iteration.